### ISSUE TITLE
--confidence modifies states

### ISSUE THREAD

**Issue by tzukuoh** [JohnLangford/vowpal_wabbit#885]
./vw [train.txt](https://github.com/JohnLangford/vowpal_wabbit/files/63246/train.txt)

Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = train
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
1.000000 1.000000            1            1.0   1.0000   0.0000        4
0.589811 0.179622            2            2.0   1.0000   0.5762        4
1.294905 2.000000            4            4.0  -1.0000   1.0000        4

finished run
number of examples per pass = 5
passes used = 1
weighted example sum = 5.000000
weighted label sum = 1.000000
average loss = 1.279486
best constant = 0.200000
best constant's loss = 0.960000
total feature number = 20

./vw --confidence train.txt

Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = train
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
confidence = 0 pred = 0 threshold = 0 sensitivity = 0.499934
1.000000 1.000000            1            1.0   1.0000   0.0000        4
confidence = 1.43092 pred = 0.576182 threshold = 0 sensitivity = 0.40625
0.589811 0.179622            2            2.0   1.0000   0.5762        4
confidence = 1.47462 pred = 1 threshold = 0 sensitivity = 0.674957
confidence = 4.61897 pred = 1 threshold = 0 sensitivity = 0.217359
1.294905 2.000000            4            4.0  -1.0000   1.0000        4
confidence = 1.579 pred = 0.191733 threshold = 0 sensitivity = 0.123352

finished run
number of examples per pass = 5
passes used = 1
weighted example sum = 5.000000
weighted label sum = 1.000000
average loss = 1.319970
best constant = 0.200000
best constant's loss = 0.960000
total feature number = 20

Using --confidence results in a different loss.

**Comment by tzukuoh** [comment]
I looked into gd.cc and found that in get_pred_per_update, when normalized  is true 
g.all->normalized_sum_norm_x and g.total_weight get modified even when stateless is true.
That looks like an issue?

**Comment by tzukuoh** [comment]
This is fixed by merged pull request #886.
